satpy/tests/reader_tests/test_msu_gsa_l1b.py

```diff
@@ -0,0 +1,180 @@
+                    da.ones((num_scans*4, num_cols*4), chunks=1024,
+                            dtype=np.uint16),
+                    attrs={
+                        'scale': 0.01, 'offset': 0., 'fill_value': -999.
+                    },
+                    dims=('x', 'y')),
+            'Data/resolution_1km/Radiance_01':
+                xr.DataArray(
+                    da.ones((num_scans*4, num_cols*4), chunks=1024,
+                            dtype=np.uint16),
+                    attrs={
+                        'scale': 0.01, 'offset': 0., 'fill_value': -999., 'F_solar_constant': SOLCONST
+                    },
+                    dims=('x', 'y')),
+            'Data/resolution_4km/Solar_Zenith_Angle':
+                xr.DataArray(
+                    da.ones((num_scans, num_cols), chunks=1024,
+                            dtype=np.uint16),
+                    attrs={
+                        'scale': 0.01, 'offset': 0., 'fill_value': -999.
+                    },
+                    dims=('x', 'y')),
+            'Geolocation/resolution_4km/Latitude':
+                xr.DataArray(
+                    da.ones((num_scans, num_cols), chunks=1024,
+                            dtype=np.uint16),
+                    attrs={
+                        'scale': 0.01, 'offset': 0., 'fill_value': -999.
+                    },
+                    dims=('x', 'y')),
+            'Geolocation/resolution_4km/Longitude':
+                xr.DataArray(
+                    da.ones((num_scans, num_cols), chunks=1024,
+                            dtype=np.uint16),
+                    attrs={
+                        'scale': 0.01, 'offset': 0., 'fill_value': -999.
+                    },
+                    dims=('x', 'y')),
+            'Data/resolution_4km/Brightness_Temperature_09':
+                xr.DataArray(
+                    da.ones((num_scans, num_cols), chunks=1024,
+                            dtype=np.uint16),
+                    attrs={
+                        'scale': 0.01, 'offset': 0., 'fill_value': -999.
+                    },
+                    dims=('x', 'y')),
+        }
+        return data
+
+    def get_test_content(self, filename, filename_info, filetype_info):
+        """Mimic reader input file content."""
+        num_scans = 20
+        num_cols = 2048
+        global_attrs = {
+            '/attr/timestamp_without_timezone': '2022-01-13T12:45:00',
+            '/attr/satellite_observation_point_height': '38500.0',
+            '/attr/satellite_observation_point_latitude': '71.25',
+            '/attr/satellite_observation_point_longitude': '21.44',
+        }
+
+        data = self._get_data(num_scans, num_cols)
+
+        test_content = {}
+        test_content.update(global_attrs)
+        test_content.update(data)
+        return test_content
+
+
+class TestMSUGSABReader(unittest.TestCase):
+    """Test MSU GS/A L1B Reader."""
+
+    yaml_file = "msu_gsa_l1b.yaml"
+
+    def setUp(self):
```

Comment:
```suggestion
class TestMSUGSABReader:
    """Test MSU GS/A L1B Reader."""

    yaml_file = "msu_gsa_l1b.yaml"

    def setup(self):
```

Reply:
That is we don't need to use the `unittest` framework that what we use pytest

Reply:
Sorry, I don't quite understand the logic of removing the `unittest.TestCase` here, I can replace `self.assertTrue` with `assert` but can't replace `assertIn` or `assertEqual`, so would have to import unittest anyway, which seems to defeat the point.

Reply:
Ok, sorry for the lack of explanation :) the idea is to do `assert` instead of `self.assertTrue`, then `assert something in something_bigger` instead of `self.assertIn`, etc. Pytest will catch potential errors and format them correctly, while still allowing regular python syntax to be used for the assertions, instead of `unittest`'s object-oriented syntax.

Reply:
Also, if I change `setUp` to `setup` then the setup functions don't get run any more.

Reply:
they should??? I have an example of reader tests using pytest here if it helps: https://github.com/pytroll/satpy/pull/1981/files#diff-5d6073c04cf62a8a99f740c2824b24ce4dba92d17d642e2893096f4b9884db45R85

Reply:
I usually use `setup_method` and `teardown_method` or there are class versions too.

Reply:
Not sure why it wasn't working before, looks like I made a mistake. Tests passing now!